pkg/controllers/user/ingress/ingress.go

```
@@ -86,13 +71,37 @@ func (c *Controller) sync(key string, obj *v1beta1.Ingress) error {
 		}
 	}
 
+	ingressServices := map[string]*corev1.Service{}
+	services, err := c.serviceLister.List(obj.Namespace, labels.NewSelector())
+	if err != nil {
+		logrus.Error(err)
+		services = []*corev1.Service{}
```

we should throw an error all the way up, do not default to empty